.github/workflows/iroha2-dev-pr.yml

```diff
@@ -95,6 +95,61 @@ jobs:
           # This context specification is required
           context: .
 
+  workspace_analysis_clippy:
+    runs-on: ubuntu-latest
+    container:
+      image: hyperledger/iroha2-ci:nightly-2024-04-18
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          repository: ${{ github.event.workflow_run.head_repository.full_name }}
+          ref: ${{ github.event.workflow_run.head_branch }}
+      - uses: Swatinem/rust-cache@v2
+      - name: Format
+        run: cargo fmt --all -- --check
+      - name: Lints without features
+        if: always()
+        run: cargo clippy --workspace --benches --tests --examples --no-default-features --quiet
+      - name: Lints with all features enabled
+        if: always()
+        run: cargo clippy --workspace --benches --tests --examples --all-features --quiet --message-format=json | tee clippy.json
+      - name: Documentation
+        if: always()
+        run: cargo doc --no-deps --quiet
+      - name: Upload clippy report artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: clippy.json
+          path: clippy.json
+
+  # exclude: client/tests/integration/
+  with_coverage:
+    runs-on: [self-hosted, Linux, iroha2]
+    container:
+      image: hyperledger/iroha2-ci:nightly-2024-04-18
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          repository: ${{ github.event.workflow_run.head_repository.full_name }}
+          ref: ${{ github.event.workflow_run.head_branch }}
+      - uses: Swatinem/rust-cache@v2
+      - name: Run tests, with coverage
+        run: |
+          mold --run cargo test --all-features --no-fail-fast --workspace --exclude iroha
+          mold --run cargo test --all-features --no-fail-fast -p iroha -- --skip integration
```

Comment:
I think we also used to have tests for no-default-features

Reply:
Are n't they in `integration` and `extra_functional` jobs?
https://github.com/hyperledger/iroha/blob/2200ee9a8c76faa4d6a893806cf26988bc0ebe40/.github/workflows/iroha2-dev-pr.yml#L76

Reply:
Nope, these are integration tests

Reply:
> I think we also used to have tests for no-default-features

@s8sato your opinion?

Reply:
it was removed in this [PR](https://github.com/hyperledger/iroha/pull/3037)